wordpress-mobile/WordPress-Android
# [Reader] Implement tags feed card actions (part 2)

Fixes #20670 

Part 2 of the "Tags" feed card actions implementation. Actions implemented in this PR:

- Like.

Next PR (part 3):
- Overflow menu.

-----

## To Test:

1. Instal JP and log in.
2. Enable `reader_tags_feed` feature flag and restart the app.
3. Open Reader and select "Tags" feed with at least one tag subscribed.
4. 🔍 Tap the like button in any post: the status should change right away. If the request fails, the status is reverted. It should not be possible to tap the button again while the request is running (the button is disabled until the request finishes).
5. 🔍 Tap the like button without internet connection: you should see the an error message.
6. 🔍 There's also an error message that is shown when the back-end request returns with an error, but I don't think there's a way to test it out without mocking it or using a proxy.

-----

## Regression Notes

1. Potential unintended areas of impact

    - None

2. What I did to test those areas of impact (or what existing automated tests I relied on)

    - Manual testing and updated unit tests.

3. What automated tests I added (or what prevented me from doing so)

    - Updated `ReaderTagsFeedViewModelTest.kt` and `ReaderTagsFeedUiStateMapperTest.kt`.

-----

## PR Submission Checklist:


-----

## Testing Checklist (strike-out the not-applying and unnecessary ones):
